src/modules/cpu/kernel/transpose.hpp

```diff
@@ -0,0 +1,445 @@
+                    }
+                }
+                // Optimized AVX version for 3D inputs and permutation order (0, 2, 1)
+                else if(perm[0] == 0 && perm[1] == 2 && perm[2] == 1)
+                {
+                    Rpp32f *srcPtrRow = srcPtrTemp;
+                    Rpp32f *dstPtrRow = dstPtrTemp;
+                    for(int i = 0; i < length[0]; i++)
+                    {
+                        compute_2d_transpose(srcPtrTemp, dstPtrTemp, length[1], length[2], srcGenericDescPtr->strides[2], dstGenericDescPtr->strides[2]);
+
+                        // increment src and dst pointers
+                        srcPtrTemp += srcGenericDescPtr->strides[1];
+                        dstPtrTemp += dstGenericDescPtr->strides[1];
+                    }
+                }
+                else
+                {
+                    transpose_generic_setup_and_run(srcPtrTemp, dstPtrTemp, length, perm, nDim);
+                }
+            }
+            else if (nDim == 4)
+            {
+                // Optimized AVX version for 4D inputs and permutation order (1, 2, 3, 0)
+                Rpp32u vectorIncrement = 8;
+                if(perm[0] == 1 && perm[1] == 2 && perm[2] == 3 && perm[3] == 0)
+                {
+                    Rpp32u bufferLength = length[perm[3]];
+                    Rpp32u alignedLength = (bufferLength / vectorIncrement) * vectorIncrement;
+                    Rpp32f *srcPtr0 = srcPtrTemp;
+                    Rpp32f *dstPtr0 = dstPtrTemp;
+                    for(int i = 0; i < length[perm[0]]; i++)
+                    {
+                        Rpp32f *srcPtr1 = srcPtr0;
+                        Rpp32f *dstPtr1 = dstPtr0;
+                        for(int j = 0; j < length[perm[1]]; j++)
+                        {
+                            Rpp32f *srcPtr2 = srcPtr1;
+                            Rpp32f *dstPtr2 = dstPtr1;
+                            for(int k = 0; k < length[perm[2]]; k++)
+                            {
+                                Rpp32f *srcPtr3 = srcPtr2;
+                                Rpp32f *dstPtr3 = dstPtr2;
+
+                                Rpp32u vectorLoopCount = 0;
+                                for( ; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                                {
+                                    __m256 pSrc = _mm256_setr_ps(*srcPtr3,
+                                                                 *(srcPtr3 + srcGenericDescPtr->strides[1]),
+                                                                 *(srcPtr3 + 2 * srcGenericDescPtr->strides[1]),
+                                                                 *(srcPtr3 + 3 * srcGenericDescPtr->strides[1]),
+                                                                 *(srcPtr3 + 4 * srcGenericDescPtr->strides[1]),
+                                                                 *(srcPtr3 + 5 * srcGenericDescPtr->strides[1]),
```

Comment:
This whole 1/2/3/4/5/6/7 * srcGenericDescPtr->strides[1] is constant. 
Can be taken out of 4 loops!